doc/articles/features/windows-storage-pickers.md

```diff
@@ -0,0 +1,228 @@
+// this starts the download process of the browser
+await CachedFileManager.CompleteUpdatesAsync(file);
+```
+
+### Choosing supported type of pickers
+
+By default, Uno Platform attempts to use File System Access API and falls back to download/upload pickers if not available. To control this behavior, you can use `WinRTFeatureConfiguration`:
+
+```c#
+#if __WASM__
+Uno.WinRTFeatureConfiguration.Storage.Pickers.WasmConfiguration = 
+    WasmPickerConfiguration.FileSystemAccessApiWithFallback;
+#endif
+```
+
+The allowed values for the configuration are:
+
+- `FileSystemAccessApiWithFallback` - defaults to File System Access API, but falls back to download/upload pickers if not available
+- `FileSystemAccessApi` - uses File System Access API only. If not avaialable, pickers will throw `NotSupportedException`
+- `DownloadUpload` - uses download/upload pickers only.
+
+## Android
+
+Files picked from file pickers on Android are provided by the *Storage Access Framework API*. Due to its limitations, it is not possible to write to existing file in-place. Instead, Uno Platform creates a copy of the file in temporary storage and your changes are applied to this temporary file instead. When your file stream is then flushed, closed, or disposed of, the changes are written to the source file and the temporary file is discarded.
+
+## iOS
+
+iOS does not offer a built-in `FileSavePicker` alternative. Luckily it is possible to implement this functionality for example using a combination of a `FolderPicker` and `ContentDialog`.
+
+To provide your own custom implementation, create a class that implements the `IFileSavePickerExtension` which is only available on iOS. This class must have a `public` constructor with a `object` parameter. This will actually be an instance of `FileSavePicker` when invoked later. Then implement the `PickSaveFileAsync` method:
+
+```c#
+#if __IOS__
+namespace Custom.Pickers
+{
+    public class CustomFileSavePickerExtension : IFileSavePickerExtension
+    {
+        private readonly FileSavePicker _fileSavePicker;
+    
+        public CustomFileSavePickerExtension(object owner)
+        {
+            _fileSavePicker = (FileSavePicker)picker;
+        }
+        
+        public async Task<StorageFile> PickSaveFileAsync(CancellationToken token)
+        {
+            // ... your own implementation
+        }
+    }
+}
+#endif
+```
+
+When done, register this extension in `App.xaml.cs`:
+
+```c#
+public App()
+{
+#if __IOS__
+    ApiExtensibility.Register(
+        typeof(Uno.Extensions.Storage.Pickers.IFileSavePickerExtension), 
+        (picker) => new CustomFileSavePickerExtension(picker));
+#endif
+}
+```
+
+As this is quite complex, you can check a full sample implementation of a folder-based save file picker <a href="https://github.com/unoplatform/Uno.Samples/pull/48" target="_blank">here in the Uno.Samples repository</a>. You can modify and adjust this implementation as you see fit for your specific use case.
```

Comment:
Maybe reference the sample directory directly, so it remains up-to-date.
```suggestion
As this is quite complex, you can check a full sample implementation of a folder-based save file picker: [Uno.Samples - FileSavePickeriOS](https://github.com/unoplatform/Uno.Samples/tree/master/UI/FileSavePickeriOS). You can modify and adjust this implementation as you see fit for your specific use case.
```
note: _of course the above wont work until merged, so here is [the same link from your repo](https://github.com/MartinZikmund/Uno.Samples/tree/dev/mazi/file-save-picker/UI/FileSavePickeriOS)_